usb/usb_host_msc/src/msc_host.c

```diff
@@ -320,17 +322,34 @@ static bool is_mass_storage_device(uint8_t dev_addr)
     return is_msc_device;
 }
 
-static void event_handler_task(void *arg)
+
+
+esp_err_t msc_host_handle_events(uint32_t timeout)
 {
-    while (1) {
-        usb_host_client_handle_events(s_msc_driver->client_handle, portMAX_DELAY);
+    MSC_RETURN_ON_FALSE(s_msc_driver != NULL, ESP_ERR_INVALID_STATE);
 
-        if (s_msc_driver->end_client_event_handling) {
-            break;
-        }
+    ESP_LOGV(TAG, "USB MSC handling");
+    s_msc_driver->event_handling_started = true;
```

Comment:
Here is the question: what if the MSC driver was configured with bg_task=false, but the `msc_host_handle_events` has never been called? 
Seems that `event_handling_started` flag will be staying false and semaphore does not let the `msc_host_uninstall` function be successfully completed. 

Does that make sense? :thinking:

Reply:
Yes, it does. I will extend the tests to cover this situation

Reply:
Hi @roma-jam I extended the tests with the scenario you described, all tests passing!

Reply:
Hi @tore-espressif ,

Yes, I did the same on HID and that also works.